File: web_app/src/main/java/br/edu/usf/ads/web/models/Aluno.java
package br.edu.usf.ads.web.models;

import org.jetbrains.annotations.NotNull;

import java.sql.ResultSet;
import java.sql.SQLException;

public class Aluno {

    private int id;
    private String ra;
    private String nome;

    @org.jetbrains.annotations.Contract(pure = true)
    public Aluno() {
        super();
    }

    public Aluno(int id, String ra, String nome) {
        setId(id);
        setRa(ra);
        setNome(nome);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getRa() {
        return ra;
    }

    public void setRa(String ra) {
        this.ra = ra;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    @NotNull
    public static Aluno fromDAO(@NotNull ResultSet result) throws SQLException {
        int id = result.getInt("id");
        String ra = result.getString("ra");
        String nome = result.getString("nome");

        return new Aluno(id, ra, nome);
    }

    @NotNull
    public String tableRow() {
        return "<tr>" + "<th scope=\"row\">" + id + "</th>" +
                "<td>" + ra + "</td>" +
                "<td>" + nome + "</td>" +
                "<td>" +
                "<span data-id=\"" + id + "\" data-ra=\"" + ra + "\" data-nome=\"" + nome + "\" class=\"table-edit\"><a href=\"#editModal\" class=\"edit\" data-toggle=\"modal\"><i class=\"material-icons\" data-toggle=\"tooltip\" title=\"Editar\">&#xE254;</i></a></span>" +
                "<span data-id=\"" + id + "\" data-ra=\"" + ra + "\" data-nome=\"" + nome + "\" class=\"table-remove\"><a href=\"#deleteModal\" class=\"delete\" data-toggle=\"modal\"><i class=\"material-icons\" data-toggle=\"tooltip\" title=\"Deletar\">&#xE872;</i></a></span>" +
                "</td>" +
                "</tr>";
    }
}

File: web_service/models/aluno.php
<?php
    require_once 'exposable.php';

    class Aluno implements Exposable {

        private $id;
        private $ra;
        private $nome;
        
        function __contruct($id, $ra, $nome) {
            setId($id);
            setRa($ra);
            setNome($nome);
        }
        
        function getId() {
            return $this->id;
        }
        
        function setId($id) {
            $this->id = $id;
        }
        
        function getRa() {
            return $this->ra;
        }

        function setRa($ra) {
            $this->ra = $ra;
        }

        function getNome() {
            return $this->nome;
        }
        
        function setNome($nome) {
            $this->nome = $nome;
        }

        function expose() {
            return get_object_vars($this);
        }
        
        function printInfo() {
            echo '<br>---------------[ALUNO]---------------';
            echo '<br>ID: '.$this->getId();
            echo '<br>RA: '.$this->getRa();
            echo '<br>Nome: '.$this->getNome();
            echo '<br>';
        }
        
        public static function fromPDO($fPDO) {
            $student = new Aluno();
            $student->setId($fPDO['id']);
            $student->setRa($fPDO['ra']);
            $student->setNome($fPDO['nome']);
        
            return $student;
        }
    }
?>